### ISSUE TITLE
REGEX recognition

### ISSUE THREAD

**Issue by phajdan** [hunspell/hunspell#275]
Hello,

It could be useful to recognize words by regular expression.

Exemple :

REGEX 1
REGEX [a-z0-9-]+[.](org|net|com|hu|fr|uk)$ po:URL

Original comment by: dicollecte

Original Ticket: [hunspell/feature-requests/34](https://sourceforge.net/p/hunspell/feature-requests/34)

**Comment by baarsrj** [comment]
Yes, this would help underlines certain types of 'words' for which spellchecking is not relevant at all.
This way, numbers could be excluded and CamelBack words, being proper names.

**Comment by Kruno-S** [comment]
I know this thread was opened six years ago, but since it has been commented recently, I think it's better to post here then to open a new thread on pretty much the same topic. I'm not commenting or referring to anyone’s answer or comment, just presenting my observations.

First of all, think it's wrong to try to make spelling checker to stop underlining certain 'words' like URLs, license plates, roman numerals and such. Underlining is not bad – it's a tool and should not be used sparingly.

Since license plates can't be inserted into a dictionary like normal words, of course they are underlined. But the cache is in that user can misspell that too. There's no point of using regular expressions _not to underline_ such text strings (can't really call them a words). If string is not underlined, user will take its correctness for granted. Spelling checker can't confirm if user wanted to write 'XI' (eleven) or 'IX' (nine) or even 'XII' or 'IXX'. Yes, it's not that simple but spelling checker is useless on such examples and it can't effectively proofread this and do it with grammar checker is hard also.

It's really (really!) wrong to make spelling checker not to underline stuff it can't confirm or deny. It's like saying 'I have no idea what you just wrote – but it's correct, rock on!'.

So what I would suggest is to make ignore file: xx_XX.igr and make it work with affix file _and_ regular expressions. What's in this file should be underlined yellow, orange or such to signal user that spell checker surrendered so user should proofread this on it's own. It's not about being correct or incorrect, it's just something spell checker denies to check – and user should know that.

I know this is ambitious but can’t regular expressions at least be used in affix file _to underline_ URLs, license plates, maybe roman numerals and stuff like that, but just in a different color?

Can't be good to stop underlining just for sake of it.

I really hope that someone will comment on this :)
